src/utils/resolveThunk.ts

```diff
@@ -0,0 +1,8 @@
+import { Thunk } from 'graphql';
+
+function isFunction<T>(fn: unknown): fn is () => T {
```

Comment:
Had to add this type guard to fix the error

Reply:
can you not inline the `typeof` check below? what is the type for `Thunk<T>` that it doesn't narrow

Reply:
alt try `thunk instanceof Function`